Properly check that reduction strings are valid for l1_loss, smoothl1_loss, and mse_loss.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43809 Actually run backward criterion tests.
* #43788 Merge criterion and new criterion tests.
* **#43527 Properly check that reduction strings are valid for l1_loss, smoothl1_loss, and mse_loss.**
* #43523 Kill unused _pointwise_loss function.

Differential Revision: [D23306786](https://our.internmc.facebook.com/intern/diff/D23306786)

gchanan committed:
Properly check that reduction strings are valid for l1_loss, smoothl1_loss, and mse_loss.

gchanan commented:
TODO: testing

gchanan committed:
Update on "[WIP] Properly check that reduction strings are valid for l1_loss, smoothl1_loss, and mse_loss."


Differential Revision: [D23306786](https://our.internmc.facebook.com/intern/diff/D23306786)

gchanan committed:
Update on "[WIP] Properly check that reduction strings are valid for l1_loss, smoothl1_loss, and mse_loss."


Differential Revision: [D23306786](https://our.internmc.facebook.com/intern/diff/D23306786)

[ghstack-poisoned]

albanD reviewed (commented):
LGTM

`F.ctc_loss()` is not tested here, is that on purpose?

gchanan committed:
Update on "Properly check that reduction strings are valid for l1_loss, smoothl1_loss, and mse_loss."


Differential Revision: [D23306786](https://our.internmc.facebook.com/intern/diff/D23306786)

gchanan commented:
good point, I'll add `ctc_loss`.
